modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Tracker.scala

```diff
@@ -81,26 +74,30 @@ class Tracker(emitters: Seq[TEmitter], namespace: String, appId: String, encodeB
    */
   private def completePayload(payload: Payload,
                               contexts: Seq[SelfDescribingJson],
-                              timestamp: Option[Timestamp]): Payload = {
-
-    payload.add("eid", UUID.randomUUID().toString)
-
-    if (!payload.nvPairs.contains("dtm")) {
-      timestamp match {
-        case Some(DeviceCreatedTimestamp(dtm)) => payload.add("dtm", dtm.toString)
-        case Some(TrueTimestamp(ttm))          => payload.add("ttm", ttm.toString)
-        case None                              => payload.add("dtm", System.currentTimeMillis().toString)
+                              timestamp: Option[Timestamp]): F[Payload] =
+    for {
+      uuid   <- emitters.head.generateUUID
+      millis <- emitters.head.getCurrentMilliseconds
```

Comment:
this makes it apparent that uuid and ms don't belong in the emitter algebra, no?

Reply:
@BenFradet what do you propose?

Reply:
can we have independent algebras? that'll make the transition easier since they're being integrated into cats-effect?

Reply:
That was Anton's idea also, I thought I would simplify it by merging it with the emitter algebra. You are absolutely right though, logged #99